components/memory/component.go

```diff
@@ -5,42 +5,62 @@ import (
 	"context"
 	"database/sql"
 	"fmt"
-	"strconv"
 	"time"
 
 	"github.com/leptonai/gpud/components"
-	"github.com/leptonai/gpud/components/common"
-	"github.com/leptonai/gpud/components/dmesg"
+	events_db "github.com/leptonai/gpud/components/db"
 	memory_id "github.com/leptonai/gpud/components/memory/id"
 	"github.com/leptonai/gpud/components/memory/metrics"
 	"github.com/leptonai/gpud/components/query"
-	query_log "github.com/leptonai/gpud/components/query/log"
 	"github.com/leptonai/gpud/log"
 
 	"github.com/prometheus/client_golang/prometheus"
 )
 
-func New(ctx context.Context, cfg Config) components.Component {
+func New(ctx context.Context, cfg Config) (components.Component, error) {
 	cfg.Query.SetDefaultsIfNotSet()
 	setDefaultPoller(cfg)
 
 	cctx, ccancel := context.WithCancel(ctx)
 	getDefaultPoller().Start(cctx, cfg.Query, memory_id.Name)
 
-	return &component{
-		rootCtx: ctx,
-		cancel:  ccancel,
-		poller:  getDefaultPoller(),
+	eventsStore, err := events_db.NewStore(
+		cfg.Query.State.DBRW,
+		cfg.Query.State.DBRO,
+		events_db.CreateDefaultTableName(memory_id.Name),
+		3*24*time.Hour,
+	)
+	if err != nil {
+		ccancel()
```

Comment:
it is a bit strange to cancel on error. Do we really need to create a `cctx, ccancel := context.WithCancel(ctx)`? Can we just use ctx?

Reply:
creating event store first, so we don't need call `ccancel`

Reply:
@cardyok do you have any idea how to make the ctx usage look better?